llvm/lib/BinaryFormat/DXContainer.cpp

```diff
@@ -89,3 +89,33 @@ static const EnumEntry<PSV::InterpolationMode> InterpolationModeNames[] = {
 ArrayRef<EnumEntry<PSV::InterpolationMode>> PSV::getInterpolationModes() {
   return ArrayRef(InterpolationModeNames);
 }
+
+#define RESOURCE_TYPE(Val, Enum) {#Enum, PSV::ResourceType::Enum},
+
+static const EnumEntry<PSV::ResourceType> ResourceTypeNames[] = {
+#include "llvm/BinaryFormat/DXContainerConstants.def"
+};
```

Comment:
Nit: Style-wise, we usually put the `#define` immediately before `#include` of a def file, like so:
```suggestion
static const EnumEntry<PSV::ResourceType> ResourceTypeNames[] = {
#define RESOURCE_TYPE(Val, Enum) {#Enum, PSV::ResourceType::Enum},
#include "llvm/BinaryFormat/DXContainerConstants.def"
};
```

I find this makes it more obvious that the two things are connected. This suggestion also applies to the enum definitions in the header.

Reply:
I was about to give similar feedback in that llvm/include/llvm/BinaryFormat/DXContainer.h but it pretty consistently puts the `#define` before the enum.

I wonder if a fix here, or maybe in a follow up, to update DXContainer to this way would be worth doing?